scipy/optimize/tests/test_quadratic_assignment.py

```diff
@@ -429,3 +440,10 @@ def _doubly_stochastic(P, tol=1e-3):
         P_eps = r[:, None] * P * c
 
     return P_eps
+
+
+def _er_matrix(n, p):
+    x = np.triu(np.random.rand(n, n))
+    m = x + x.T
+    m[range(n), range(n)] = 0
```

Comment:
```suggestion
    np.fill_diagonal(m, 0)
```

Reply:
If you take the triu with `k=1` then you don't need to reset the diagonals.

Reply:
Ah good call.